rest/src/main/java/cz/idomatojde/rest/controllers/UserController.java

```diff
@@ -50,6 +52,11 @@ public UserController(UserFacade users, OfferFacade offers) {
         return ok(facade.getUserContactInfo(userId));
     }
 
+    @PostMapping(value = "signup")
+    ResponseEntity<Long> signup(@RequestBody RegisterUserDTO regDto) {
```

Comment:
I would modify the base class and use the `regsiter` function there (another `final boolean allowUnauthenticatedRegister` perhaps? and set it only for the `UserController`)?